hnswlib/space_ip.h

```diff
@@ -283,6 +292,19 @@ namespace hnswlib {
         InnerProductSpace(size_t dim) {
             fstdistfunc_ = InnerProduct;
     #if defined(USE_AVX) || defined(USE_SSE) || defined(USE_AVX512)
+        #if defined(USE_AVX512)
+            if (AVX512Capable())
+                InnerProductSIMD16Ext = InnerProductSIMD16ExtAVX512;
+            else if (AVXCapable())
+                InnerProductSIMD16Ext = InnerProductSIMD16ExtAVX;
+        #endif
+        #if defined(USE_AVX)
```

Comment:
I guess here should be `#elif`

Reply:
Ah yeah.  I tried to combine the check for both 16Ext and 4Ext.  But because 4Ext doesn't use AVX512 that led to a mistake.  If you do implement AVX512 in the future for 4Ext, then they can be combined in the runtime check.  I'll fix that and trigger another review.